concrete/src/Tree/Node/Type/FileFolder.php

```diff
@@ -31,6 +47,104 @@ public function getTreeNodeTypeName()
         return t('Folder');
     }
 
+    public function getTreeNodeDisplayName($format = 'html')
+    {
+        if ($this->getTreeNodeName()) {
+            $name = tc($this->getTreeNodeTranslationContext(), $this->getTreeNodeName());
+            switch ($format) {
+                case 'html':
+                    return h($name);
+                case 'text':
+                default:
+                    return $name;
+            }
+        } elseif ($this->treeNodeParentID == 0) {
+            return t('Folders');
+        }
+    }
+
+    /**
+     * Returns the storage location id of the folder.
+     *
+     * @return int|null
+     */
+    public function getTreeNodeStorageLocationID()
+    {
+        return (int) $this->fslID;
+    }
+
+    /**
+     * @return \Concrete\Core\Entity\File\StorageLocation\StorageLocation|null
+     */
+    public function getTreeNodeStorageLocationObject()
+    {
+        $app = Application::getFacadeApplication();
+
+        return $app->make(StorageLocationFactory::class)->fetchByID((int) $this->fslID);
```

Comment:
Maybe use the method as well and get rid of the type cast? Also, according to https://github.com/concrete5/concrete5/pull/7344/files#diff-5e157b1f79527aee7488ec3e9fdd3bd2R21 this column should never contain null, and is always an integer?